analyzers/src/SonarAnalyzer.CSharp/Rules/DoNotCallExitMethods.cs

```diff
@@ -48,6 +48,7 @@ public sealed class DoNotCallExitMethods : DoNotCallMethodsCSharpBase
             !invocationSyntax
                 .Ancestors()
                 .OfType<BaseMethodDeclarationSyntax>()
+                .Where(x => x.GetIdentifierOrDefault()?.ValueText == "Main")
```

Comment:
I was curious how this works for top-level statements and it looks like it doesn't. Shouldn't this be a false positive: https://github.com/SonarSource/sonar-dotnet/blob/master/analyzers/tests/SonarAnalyzer.UnitTest/TestCases/DoNotCallExitMethods.CSharp9.cs#L3 ?

Reply:
It was a really small change, so I fixed it in this PR. I hope you don't mind.